### DISCUSSION TITLE
Unreliable on arm64?

### DISCUSSION THREAD

**Discussion by dmindfuse** [dani-garcia/vaultwarden#1614] [Show and tell]
Hello community. Having been a paying customer for a password manager service I was intrigued by Bitwarden_RS. Could I really operate a password manager to the level of reliability and availability provided by a pay-for service? So I set out to install it following the instructions online. I have deployed it on a Rock Pi 64 (arm64v8). This tiny hobby device also consumes between 50% and 80% of its CPU for BOINC projects, Portainer, Kanboard. Lo and behold, a few clicks and some configuration changes and restarts and there it is: your own bitwarden_rs instance. I have to say, I was very impressed. It even includes fully configured HTTPS out of the box. Even 2FA worked, E-mail delivery, you name it. I was absolutely genuinely impressed.

So what's not to like? Well, so far, runing it for 3 days t's been real hit and miss with the server's responsiveness. I have to frequently restart both Caddy and the Bitwarden containers to fix things like:
- http2 stream issues reported in the log file
- the web vault login page stuck on a rotating wheel infinitely (all browsers)
- performance: deleting 300 items from the vault caused the web vault to become unresponsive (sqlite3?)
- sync on my iOS clients hanging/unresponsive and blocking the app in the process

What's it been for others on aarch64?

**Comment by pepa65** [comment]
Don't know its specs, but it does sound you're already stretching the device even before running Bitwarden. Curious now as to other people's experiences on lower powered devices.

**Reply by dmindfuse** [reply]
I have since abandoned this. Not because of the device or performance, but because backing up and managing my passwords was going to cost me more than what I pay for a managed password manager App annually. I know I know… security, privacy, trust… still… 😑 thanks though.

**Reply by BlackDex** [reply]
What is since? The current `testing` tagged images are totally different right now.
So posting a comment like this is a bit strange.

**Reply by dmindfuse** [reply]
@BlackDex , sorry if I caused confusion. That was not intentional. I meant to say that I have abandoned testing the application _since last I wrote_ here. I thought I might close this thread, but that appears to be impossible for me. For the record, Bitwarden is genuinely impressive, but I haven't -personally- the time to invest into setting things up securely and resiliently (I'm no spring chicken...) :)

**Reply by BlackDex** [reply]
Ah, thanks for the clarification :smile:, and since this is a discussion topic it can't be closed.
A lot of changes have been done in the mean time, and i know there are some other changes we can make to increase some performance on some tasks which i haven't gotten to as of yet, i just want to make sure that people understand that as with all software it is a work in progress, and sometimes optimizing an application can take some time.
